File: app/src/main/java/com/tk4218/grocerylistr/CustomLayout/CalendarAdapter.java
package com.tk4218.grocerylistr.CustomLayout;

import android.content.Context;
import android.graphics.Color;
import android.util.Log;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TextView;

import com.tk4218.grocerylistr.Model.CalendarRecipes;
import com.tk4218.grocerylistr.R;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/*
 * Created by taylo on 12/10/2017.
 */

public class CalendarAdapter extends BaseAdapter {

    private Context mContext;
    private ArrayList<CalendarRecipes> mDaysOfMonth;
    private int mCurrentMonth;
    private final SimpleDateFormat dayFormat = new SimpleDateFormat("dd", Locale.getDefault());

    public CalendarAdapter(Context context, ArrayList<CalendarRecipes> daysOfMonth, int currentMonth) {
        mContext = context;
        mDaysOfMonth = daysOfMonth;
        mCurrentMonth = currentMonth;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public int getCount() {
        return mDaysOfMonth.size();
    }

    @Override
    public CalendarRecipes getItem(int position) {
        return mDaysOfMonth.get(position);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            if(inflater == null){
                return null;
            }
            convertView = inflater.inflate(R.layout.gridview_calendar_day, parent, false);
        }

        LinearLayout calendarDay = convertView.findViewById(R.id.list_calendar_day);
        TextView dayOfMonth = convertView.findViewById(R.id.day_of_month);
        dayOfMonth.setText(dayFormat.format(mDaysOfMonth.get(position).getCalendarDate()));
        ListView dayMeals = convertView.findViewById(R.id.listview_meals);
        dayMeals.setAdapter(new CalendarDayAdapter(mContext, mDaysOfMonth.get(position).getCalendarRecipes(), mDaysOfMonth.get(position).getCalendarDate()));

        Calendar calendar = Calendar.getInstance();
        calendar.setTime(mDaysOfMonth.get(position).getCalendarDate());
        if(calendar.get(Calendar.MONTH) != mCurrentMonth){
            dayOfMonth.setTextColor(Color.LTGRAY);
        } else {
            dayOfMonth.setTextColor(Color.BLACK);
        }


        calendar.setTime(new Date());
        calendar.set(Calendar.HOUR_OF_DAY, 0);
        calendar.set(Calendar.MINUTE, 0);
        calendar.set(Calendar.SECOND, 0);
        calendar.set(Calendar.MILLISECOND, 0);

        if(mDaysOfMonth.get(position).getCalendarDate().equals(calendar.getTime())){
            calendarDay.setBackgroundResource(R.drawable.bg_current_day);
        }

        ViewGroup.LayoutParams params = convertView.getLayoutParams();
        TypedValue tv = new TypedValue();
        mContext.getTheme().resolveAttribute(android.R.attr.actionBarSize, tv, true);
        int actionBarHeight = mContext.getResources().getDimensionPixelSize(tv.resourceId);
        params.height = ((parent.getHeight() - actionBarHeight) / 6);
        convertView.setLayoutParams(params);
        return convertView;
    }
}

File: app/src/main/java/com/tk4218/grocerylistr/customlayout/CalendarAdapter.kt
package com.tk4218.grocerylistr.customlayout

import android.content.Context
import android.graphics.Color
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import android.widget.ListView
import android.widget.TextView

import com.tk4218.grocerylistr.model.CalendarRecipes
import com.tk4218.grocerylistr.R

import java.text.SimpleDateFormat
import java.util.ArrayList
import java.util.Calendar
import java.util.Date
import java.util.Locale

/*
 * Created by taylor on 12/10/2017.
 */

class CalendarAdapter(private val mContext: Context?, private val mDaysOfMonth: ArrayList<CalendarRecipes>, private val mCurrentMonth: Int) : BaseAdapter() {
    private val dayFormat = SimpleDateFormat("dd", Locale.getDefault())

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getCount(): Int {
        return mDaysOfMonth.size
    }

    override fun getItem(position: Int): CalendarRecipes {
        return mDaysOfMonth[position]
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var view = convertView
        if (view == null) {
            val inflater = mContext?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.gridview_calendar_day, parent, false)
        }

        val calendarDay = view!!.findViewById<LinearLayout>(R.id.list_calendar_day)
        val dayOfMonth = view.findViewById<TextView>(R.id.day_of_month)
        dayOfMonth.text = dayFormat.format(mDaysOfMonth[position].calendarDate)
        val dayMeals = view.findViewById<ListView>(R.id.listview_meals)
        dayMeals.adapter = CalendarDayAdapter(mContext, mDaysOfMonth[position].recipes, mDaysOfMonth[position].calendarDate!!)

        val calendar = Calendar.getInstance()
        calendar.time = mDaysOfMonth[position].calendarDate
        if (calendar.get(Calendar.MONTH) != mCurrentMonth) {
            dayOfMonth.setTextColor(Color.LTGRAY)
        } else {
            dayOfMonth.setTextColor(Color.BLACK)
        }

        calendar.time = Date()
        calendar.set(Calendar.HOUR_OF_DAY, 0)
        calendar.set(Calendar.MINUTE, 0)
        calendar.set(Calendar.SECOND, 0)
        calendar.set(Calendar.MILLISECOND, 0)

        if (mDaysOfMonth[position].calendarDate == calendar.time) {
            calendarDay.setBackgroundResource(R.drawable.bg_current_day)
        }

        val params = view.layoutParams
        val tv = TypedValue()
        mContext?.theme?.resolveAttribute(android.R.attr.actionBarSize, tv, true)
        val actionBarHeight = mContext?.resources?.getDimensionPixelSize(tv.resourceId) ?: 0
        params.height = (parent.height - actionBarHeight) / 6
        view.layoutParams = params
        return view
    }
}
